google-cloud-storage/lib/google/cloud/storage/hmac_key.rb

```diff
@@ -0,0 +1,312 @@
+        #
+        # @return [String]
+        #
+        def service_account_email
+          @gapi.service_account_email
+        end
+
+        ##
+        # The state of the key. Can be one of `ACTIVE`, `INACTIVE`, or
+        # `DELETED`.
+        #
+        # @return [String]
+        #
+        def state
+          @gapi.state
+        end
+
+        ##
+        # Last modification time of the HMAC key metadata.
+        #
+        # @return [String]
+        #
+        def updated_at
+          @gapi.updated
+        end
+
+        ##
+        # Whether the state of the HMAC key is `ACTIVE`.
+        #
+        # @return [Boolean]
+        #
+        def active?
+          state == "ACTIVE"
+        end
+
+        ##
+        # Updates the state of the HMAC key to `ACTIVE`.
+        #
+        # @return [Google::Cloud::Storage::HmacKey] Returns the HMAC key for
+        #   method chaining.
+        #
+        # @example
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
+        #
+        #   hmac_key = storage.hmac_keys.first
+        #
+        #   hmac_key.active!
+        #   hmac_key.state # "ACTIVE"
+        #
+        def active!
+          put_gapi! "ACTIVE"
+          self
+        end
+
+        ##
+        # Whether the state of the HMAC key is `INACTIVE`.
+        #
+        # @return [Boolean]
+        #
+        def inactive?
+          state == "INACTIVE"
+        end
+
+        ##
+        # Updates the state of the HMAC key to `INACTIVE`.
+        #
+        # @return [Google::Cloud::Storage::HmacKey] Returns the HMAC key for
+        #   method chaining.
+        #
+        # @example
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
+        #
+        #   hmac_key = storage.hmac_keys.first
+        #
+        #   hmac_key.inactive!
+        #   hmac_key.state # "INACTIVE"
+        #
+        def inactive!
+          put_gapi! "INACTIVE"
+          self
+        end
+
+        ##
+        # Whether the state of the HMAC key is `DELETED`.
+        #
+        # @return [Boolean]
+        #
+        def deleted?
+          state == "DELETED"
+        end
+
+        ##
+        # Deletes the HMAC key, and loads the new state of the HMAC key,
+        # which will be `DELETED`.
+        #
+        # The API call to delete the notification may be retried under certain
+        # conditions. See {Google::Cloud#storage} to control this behavior.
+        #
+        # @return [Google::Cloud::Storage::HmacKey] Returns the HMAC key for
+        #   method chaining.
+        #
+        # @example
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
+        #
+        #   hmac_key = storage.hmac_keys.first
+        #
+        #   hmac_key.inactive!.delete!
+        #
+        def delete!
+          ensure_service!
+          @service.delete_hmac_key access_id,
```

Comment:
set `project_id` parameter based on the value from `@project_id` when it's available in `HmacKey` class.

Reply:
Done, also in `#reload!`.